client/lib/user/support-user-interop.js

```diff
@@ -21,120 +18,118 @@ const STORAGE_KEY = 'boot_support_user';
  * error occurs in a wpcom API call, the error is forwarded to the
  * Redux store via an action. This also forces any data refreshes
  * that are required due to the change of user.
- *
- * @param {Object}  reduxStore  The global redux store instance
  */
-class SupportUser {
-	constructor() {
-		debug( 'Support User is enabled in this environment' );
 
-		this.reduxStoreReady = new Promise( ( resolve ) => {
-			this.setReduxStore = ( reduxStore ) => resolve( reduxStore );
-		} );
-	}
+const debug = debugModule( 'calypso:support-user' );
+const STORAGE_KEY = 'boot_support_user';
 
-	fetchToken( user, password ) {
-		debug( 'Fetching support user token' );
+export const isEnabled = () => config.isEnabled( 'support-user' );
 
-		return this.reduxStoreReady.then( ( reduxStore ) => {
-			reduxStore.dispatch( supportUserTokenFetch( user ) );
+let _setReduxStore = null;
+const reduxStoreReady = new Promise( ( resolve ) => {
+	if ( ! isEnabled() ) {
+		return;
```

Comment:
Ah, forgot to check with the feature flag off. Since this already broke, it may be worthwhile to mock in a unit test. You can test both pathways via:
```javascript
const configStub = sinon.stub( config, 'isEnabled' );
configStub.withArgs( 'support-user' ).returns( false );
```